File: Cpp/1431_Kids_With_the_Greatest_Number_of_Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> result;
        int maxValue = 0;
        for(auto candy : candies){
            maxValue = max(maxValue, candy);
        }
        for(auto candy : candies){
            if(candy + extraCandies >= maxValue){
                result.push_back(true);
            }
            else{
                result.push_back(false);
            }
        }
        return result;
    }
};

File: Java/1431_Kids_With_the_Greatest_Number_of_Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int n = candies.length;
        List<Boolean> result = new ArrayList();
        int ori_max = Integer.MIN_VALUE;
        for(int i = 0; i < n; i++){
            ori_max = Math.max(ori_max, candies[i]);
        }
        for(int i = 0; i < n; i++){
            result.add((candies[i] + extraCandies >= ori_max) ? true : false);
        }
        return result;
    }
}
